src/Microsoft.Identity.Abstractions/DownstreamRestApi/DownstreamRestApiOptions.cs

```diff
@@ -115,23 +46,26 @@ public string ProtocolScheme
         public IEnumerable<string>? Scopes { get; set; }
 
         /// <summary>
-        /// Clone the options (to be able to override them).
+        /// Optional serializer. Will serialize the input to the web API (if any).
+        /// By default, when not provided:
+        /// <list type="bullet">
+        /// <item><description>If the input derives from <c>HttpInput</c>, it's used as is</description></item>
+        /// <item><description>If the input is a string it's used as is an considered a media type json.</description></item>
+        /// <item><description>Otherwise, the object is serialized in JSON, with a UTF8 encoding, and a media
+        /// type of application/json:
+        /// <code>
+        /// new StringContent(JsonSerializer.Serialize(input), Encoding.UTF8, "application/json")
+        /// </code>
+        /// </description></item>
+        /// </list>
         /// </summary>
-        /// <returns>A clone of the options.</returns>
-        public virtual DownstreamRestApiOptions Clone()
-        {
-            return new DownstreamRestApiOptions(this);
-        }
+        public Func<object?, StringContent?>? Serializer { get; set; }
 
         /// <summary>
-        /// Return the downstream web API URL.
+        /// Optional de-serializer. Will de-serialize the output from the web API (if any).
+        /// When not provided, the following is returned:
+        /// <code>JsonSerializer.Deserialize&lt;TOutput&gt;(content, new JsonSerializerOptions { PropertyNameCaseInsensitive = true });</code>
         /// </summary>
-        /// <returns>URL of the downstream web API.</returns>
-#pragma warning disable CA1055 // Uri return values should not be strings
-        public string GetApiUrl()
-#pragma warning restore CA1055 // Uri return values should not be strings
-        {
-            return BaseUrl?.TrimEnd('/') + $"/{RelativePath}";
-        }
+        public Func<string?, object?>? Deserializer { get; set; }
```

Comment:
Why does the Deserializer take a string and the Serializer take a StringContent?

Reply:
It's what what the HttpMessage layer does ...